GoogleForCreators/web-stories-wp
# Shopping: Prevent inserting products without images

## Context

This PR disables products within the product list in shopping from being added if there is no product image, as well as an informative tooltip that explains why they cannot be added.

## Summary



## Relevant Technical Choices


`aria-disabled` was added here as an alternative to `disabled` which allows the user to focus on the item with keyboard navigation/screen reader instead of removing the button entirely from their view.

## To-do



## User-facing changes



Products without images are disabled from being added to the Story, as seen in this screenshot.
![Screen Shot 2022-05-31 at 4 55 33 PM](https://user-images.githubusercontent.com/7110244/171283279-b4ed0ce2-eae6-46fa-a69d-94c9bf4e291a.png)


## Testing Instructions






This PR can be tested by following these steps:

1. Create a new item in a tool (like WooCommerce) and make sure it has no image.
2. With the Shopping experiment enabled, open a Story.
3. Navigate to the Shopping tab and see that the button should be disabled.
4. The button should allow for you to tab-navigate to it, but cannot be added
5. The button should also show a tooltip on hover and have an aria-label with the same message as the tooltip.
6. The button, when disabled, should not allow you to add it to the Story (either by clicking or by keyboard).

To test this running locally, make sure the following is true:
- Enable the Shopping Experiment in Stories > Experiments
- Enable WooCommerce Plugin
- In Stories > Settings, make sure WooCommerce is selected as your Shopping Provider.

## Reviews

### Does this PR have a security-related impact?


No.

### Does this PR change what data or activity we track or use?


No.

### Does this PR have a legal-related impact?


No.

## Checklist




---



Fixes #11591
